File: js/discussions.js
// discussions.js
const { dumpSchema } = require("./utils/index.js");
const BraveAPI = require("./API.js");

// Create a new instance of the BraveAPI class
const brave = new BraveAPI({
    subscription_token: process.env.BRAVE_KEY
});

// Setup the search parameters
const params = new URLSearchParams({
    q: "javascript vs python",
    result_filter: "discussions"
});

// Perform the search
brave.search(params).then(results => {
    // Log the results
    console.log(dumpSchema(results.discussions));
});

File: python/discussions.py
# discussions.py
import os
from API import BraveAPI
from utils.main import dump_schema

# Create an instance of the BraveAPI class
brave = BraveAPI({
    'subscription_token': os.environ['BRAVE_KEY']
})

# Setting up parameters
params = {
    'q': "javascript vs python",
    'result_filter': "discussions",
}

# Performing the search
results = brave.search(params)

# Dumping and printing the schema of the results
print(dump_schema(results['discussions']))
